sdk/nodejs/dagger/api/api.ts

```diff
@@ -0,0 +1,127 @@
+import { Engine, GraphQLClient, gql } from "../dist/index.js";
+import { ContainerExecArgs } from "./types.js";
+import { queryBuilder, queryFlatten } from "./utils.js"
+
+type QueryTree = {
+  operation: string
+  args?: Record<string, any>
+}
+
+export class BaseApi {
+  protected _queryTree:  QueryTree[]
+  
+  constructor(queryTree: QueryTree[]= [{
+      operation: ""
+    }]) {
```

Comment:
Why not directly an empty array?

```ts
    constructor(queryTree: QueryTree[] = []) {
        this._queryTree = queryTree
    } 
```

Reply:
The file has changed a lot, this has been done on purpose at some time but you're right, a simple array is enough now. Thanks 🙏